generate.js

```diff
@@ -108,15 +116,18 @@ function next(er, input) {
 
     case 'html':
       var configDir = path.dirname(inputFile);
-      var configFile = path.basename(inputFile, '.md') + '.json';
-      configFile = path.join(configDir, configFile);
-      console.error('checking for', configFile);
-      var configObj = {};
-      if (fs.existsSync(configFile))
-        configObj = JSON.parse(fs.readFileSync(configFile))
-      require('./html.js')(input, inputFile, template, configObj, function(er, html) {
-        if (er) throw er;
-        console.log(html);
+      getFileEnding(inputFile, function(err, ending) {
+        if (err) throw err;
+        var configFile = path.basename(inputFile, ending) + '.json';
```

Comment:
Can we factor that in a separate function with a name like `getConfigForMarkdownFile`? I think it could make the code clearer.